audio/alaw.c

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file
+ * @author [sunzhenliang](https://github.com/HiSunzhenliang)
+ * @brief A-law algorithm for encoding and decoding (16bit pcm <=> a-law).
+ * @see audio/alaw.c
+ **/
+
+/**
+ * Linear input code | Compressed code | Linear output code
+ * ------------------+-----------------+-------------------
+ * s0000000abcdx     | s000abcd        | s0000000abcd1
+ * s0000001abcdx     | s001abcd        | s0000001abcd1
+ * s000001abcdxx     | s010abcd        | s000001abcd10
+ * s00001abcdxxx     | s011abcd        | s00001abcd100
+ * s0001abcdxxxx     | s100abcd        | s0001abcd1000
+ * s001abcdxxxxx     | s101abcd        | s001abcd10000
+ * s01abcdxxxxxx     | s110abcd        | s01abcd100000
+ * s1abcdxxxxxxx     | s111abcd        | s1abcd1000000
+ *
+ * https://en.wikipedia.org/wiki/G.711
+ *
+ * Compressed code: (s | eee | abcd)
+ **/
+#include <stdio.h>
+#include <stdlib.h>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c
#include <stdlib.h>     /// for IO operations
#include <assert.h>    /// for assert
```